Gems/Atom/Tools/MaterialCanvas/Assets/MaterialCanvas/TestData/ExamplePBR/example_pbr2.materialcanvasnode.azasset

```diff
@@ -0,0 +1,148 @@
+{
+    "Type": "JsonSerialization",
+    "Version": 1,
+    "ClassName": "DynamicNodeConfig",
+    "ClassData": {
+        "category": "Material Templates",
+        "title": "Example PBR 2",
+        "settings": {
+            "templatePaths": [
+                "@gemroot:MaterialCanvas@/Assets/MaterialCanvas/TestData/ExamplePBR/MaterialGraphName_ForwardPass.azsl.template",
+                "@gemroot:MaterialCanvas@/Assets/MaterialCanvas/TestData/ExamplePBR/MaterialGraphName_ForwardPass.shader.template",
+                "@gemroot:MaterialCanvas@/Assets/MaterialCanvas/TestData/ExamplePBR/MaterialGraphName.materialtype.template",
+                "@gemroot:MaterialCanvas@/Assets/MaterialCanvas/TestData/ExamplePBR/MaterialGraphName.material.template"
+            ]
+        },
+        "inputSlots": [
+            {
+                "name": "inPositionOffset",
+                "displayName": "Position Offset",
+                "description": "Position Offset",
+                "supportedDataTypes": [
+                    "float",
+                    "float2",
+                    "float3",
+                    "float4",
+                    "color"
+                ],
+                "defaultValue": {
+                    "$type": "Vector4",
+                    "Value": [
+                        0.0,
+                        0.0,
+                        0.0,
+                        0.0
+                    ]
+                },
+                "settings": {
+                    "instructions": [ "SLOTTYPE SLOTNAME = SLOTVALUE;" ]
+                }
+            },
+            {
+                "name": "inBaseColor",
+                "displayName": "Base Color",
+                "description": "Base Color",
+                "supportedDataTypes": [
+                    "float",
+                    "float2",
+                    "float3",
+                    "float4",
+                    "color"
+                ],
+                "defaultValue": {
+                    "$type": "Vector4",
+                    "Value": [
+                        1.0,
+                        1.0,
+                        1.0,
+                        1.0
+                    ]
+                },
+                "settings": {
+                    "instructions": [ "SLOTTYPE SLOTNAME = SLOTVALUE;" ]
+                }
+            },
+            {
+                "name": "inMetallic",
+                "displayName": "Metallic",
+                "description": "Metallic",
+                "supportedDataTypes": [
+                    "float",
```

Comment:
Should just be a scalar `float` I think

Reply:
As previously discussed, thinking of options to use settings for vector, scalar, matrix, and supported data types, and then auto populate this portion for graph model. For now, a lot of these basic data types support implicit conversion. There are test nodes to split and combine vector components that will also need to be revisited if we want to force explicit conversion to and from different vector and scalar types.